br: add time-zone's description of restoration (#15279)
pingcap/docs: tidb-cloud/backup-and-restore.md

### Restore data to a new cluster

> **Note**
>

> > - When you restore a DB cluster from backups, the time zone will not be overwritten by the restoration process and retain its original settings.
> >

Reviewer: ```suggestion
> When you restore a TiDB cluster from backups, the restore process retains the original time zone setting without overwriting it.
```
